src/rust/receiver-mock/src/router.rs

```diff
@@ -379,33 +374,16 @@ pub async fn handler_receiver(
 
         // Logs & events
         "application/x-www-form-urlencoded" => {
-            // TODO: refactor
-            let mut lines_count = 0 as u64;
+            {
+                let mut log_repository = app_state.logs.write().unwrap();
+                for line in lines.clone() {
+                    log_repository.add_log_message(line.to_string(), remote_address)
+                }
+            }
             if opts.print.logs {
-                for line in lines {
+                for line in lines.clone() {
```

Comment:
Are you sure we need the `clone` here?

Reply:
We don't, but it doesn't cost much, and I think it's a good practice in this kind of situation, where we have a iterator that may be consumed multiple times. Strictly speaking, the `.clone()` above is also not necessary if `opts.print_logs` is not set.